src/core/qgsvectorlayerfeaturecounter.cpp

```diff
@@ -0,0 +1,84 @@
+#include "qgsvectorlayerfeaturecounter.h"
+
+QgsVectorLayerFeatureCounter::QgsVectorLayerFeatureCounter( QgsVectorLayer *layer )
+  : mSource( new QgsVectorLayerFeatureSource( layer ) )
+  , mRenderer( layer->renderer()->clone() )
+  , mExpressionContextScopes( QgsExpressionContextUtils::globalProjectLayerScopes( layer ) )
```

Comment:
Hmmm.... 

Just thinking from a composer POV - it could be handy to be able to explicitly set the expression context here. Then the count would correctly consider the whole expression context including the atlas context...